### ISSUE TITLE
Wasmtime Async Compatibility

### ISSUE THREAD

**Issue by martindevans** [dotnet/runtime#96293]
### Description

[Wasmtime](https://wasmtime.dev/) supports async WASM execution, which can suspend running WASM code and resume it later.

Unfortunately the implementation seems to be incompatible with dotnet.

### Reproduction Steps

1. Clone [this PR](https://github.com/bytecodealliance/wasmtime-dotnet/pull/279) which I wrote for wasmtime-dotnet to add async support
2. Place a breakpoint inside an async callback
3. Run tests
4. Observe failures

### Expected behavior

Wasmtime async should work in dotnet.

Whether this should involve a change to dotnet or a change to wasmtime I'm not sure. My motivation here is really just like to draw attention to this incompatibility and to get some input from dotnet developers on the PR.

### Actual behavior

Tests fail with `System.Runtime.InteropServices.SEHException` when hitting a breakpoint.

According to [this comment](https://github.com/bytecodealliance/wasmtime-dotnet/pull/279#issuecomment-1775648516) it can sometimes crash with "Fatal error. Internal CLR error." or "AccessViolationException".

**Comment by martindevans** [comment]
I would guess arch-wasm probably isn't the right tag, since this is about using wasm within dotnet rather than dotnet within wasm.

**Comment by vcsjones** [comment]
> I would guess arch-wasm probably isn't the right tag

Yeah, I noticed - but it's probably the best way to get eyes on the issue from the right folks at this point in time - they can re-assign the arch if they feel it is incorrectly assigned.

**Comment by lewing** [comment]
More work is happening here https://github.com/dicej/isyswasfa

**Comment by pavelsavara** [comment]
I believe that we have initial implementation for this now for WASIp2. 

See
https://github.com/dotnet/runtime/blob/main/src/mono/sample/wasi/http-p2/Program.cs

This will improve with WASIp3

**Comment by martindevans** [comment]
According to [this comment](https://github.com/bytecodealliance/wasmtime-dotnet/pull/279#issuecomment-1775600594) from one of the Wasmtime developers there's no alternative way to implement stack switching on Windows others than fibers. If there's no chance of dotnet gaining support for using fibers is there some other dotnet compatible alternative wasmtime could switch to?
